packages/compass-connections-navigation/src/connection-item.tsx

```diff
@@ -0,0 +1,204 @@
+/* eslint-disable react/prop-types */
+import React, { useCallback, useMemo } from 'react';
+import {
+  useHoverState,
+  spacing,
+  css,
+  cx,
+  ItemActionControls,
+  Icon,
+} from '@mongodb-js/compass-components';
+import type { ItemAction } from '@mongodb-js/compass-components';
+import { CONNECTION_ROW_HEIGHT } from './constants';
+import {
+  ItemContainer,
+  ItemLabel,
+  ItemWrapper,
+  ItemButtonWrapper,
+} from './tree-item';
+import type {
+  VirtualListItemProps,
+  TreeItemProps,
+  NamespaceItemProps,
+} from './tree-item';
+import type { Actions } from './constants';
+import type { ConnectionInfo } from '@mongodb-js/connection-info';
+import { getItemLevelPaddingClass } from './utils';
+
+const buttonReset = css({
+  padding: 0,
+  margin: 0,
+  background: 'none',
+  border: 'none',
+});
+
+const expandButton = css({
+  display: 'flex',
+  // Not using leafygreen spacing here because none of them allow to align the
+  // button with the search bar content. This probably can go away when we are
+  // rebuilding the search also
+  padding: 7,
+  transition: 'transform .16s linear',
+  transform: 'rotate(0deg)',
+  '&:hover': {
+    cursor: 'pointer',
+  },
+});
+
+const iconStyles = css({
+  flex: 'none',
+});
+
+const expanded = css({
+  transform: 'rotate(90deg)',
+});
+
+const ExpandButton: React.FunctionComponent<{
```

Comment:
This seems to exactly copy the same code in database item. Can we move this out to the tree-item so that we don't need to maintain the same code in two places here?